_posts/2024-07-22-python-datafusion-40.0.0.md

```diff
@@ -0,0 +1,203 @@
+---
+layout: post
+title: "Apache DataFusion Python 40.0.0 Released, Significant usability updates"
+date: "2024-08-13 00:00:00"
+author: timsaucer
+categories: [release]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+## Introduction
+
+We are happy to announce that [DataFusion in Python 40.0.0] has been released. In addition to
+bringing in all of the new features of the core [DataFusion 40.0.0] package, this release
+contains *significant* updates to the user interface and documentation. We listened to the python
+user community to create a more *pythonic* experience. If you have not used the python interface to
+DataFusion before, this is an excellent time to give it a try!
+
+[DataFusion 40.0.0]: https://datafusion.apache.org/blog/2024/07/24/datafusion-40.0.0/
+[DataFusion in Python 40.0.0]: https://pypi.org/project/datafusion/40.0.0/
+
+## Background
+
+Until now, the python bindings for DataFusion have primarily been a thin layer to expose the
+underlying Rust functionality. This has been worked well for early adopters to use DataFusion
+within their Python projects, but some users have found it difficult to work with. As compared to
+other DataFrame libraries, these issues were raised:
+
+1. Most of the functions had little or no documentation. Users often had to refer to the Rust
+documentation or code to learn how to use DataFusion. This alienated some python users.
+2. Users could not take advantage of modern IDE features such as type hinting. These are valuable
+tools for rapid testing and development.
+3. Some of the interfaces felt “clunky” to users since some Python concepts do not always map well
+to their Rust counterparts.
+
+This release aims to bring a better user experience to the DataFusion Python community.
+
+## What's Changed
+
+The most significant difference is that we have added wrapper functions and classes for most of the
+user facing interface. These wrappers, written in Python, contain both documentation and type
+annotations.
+
+This documenation is now available on the [DataFusion in Python](https://datafusion.apache.org/python)
```

Comment:
Would it be better to link directly to the API reference? https://datafusion.apache.org/python/api.html